File: leetcode/0001-1000/0001-0100/0001-0010/0001-two-sum/TwoSum.java
public class TwoSum {

  public static void main(String[] args) {
    var solution = new Solution();

    System.out.println(solution.twoSum(new int[] { 2, 7, 11, 15 }, 9));
  }
}

class Solution {
  public int[] twoSum(int[] nums, int target) {
    for (var i = 0; i < nums.length; i++) {
      for (var j = i + 1; i < nums.length; j++) {
        if (nums[i] + nums[j] == target) {
          return new int[] { i, j };
        }
      }
    }

    return new int[0];
  }
}

File: leetcode/0001-1000/0001-0100/0001-0010/0001-two-sum/TwoSum.ts
function twoSum(numbers: number[], target: number): number[] {
  const cached: Record<number, number> = {};

  for (let i = 0; i < numbers.length; i++) {
    const lhs = numbers[i];
    const rhs = target - lhs;

    if (cached[rhs] !== undefined) {
      return [cached[rhs], i];
    }

    cached[lhs] = i;
  }

  return [];
};

console.log(twoSum([2, 7, 11, 15], 9)); // Output: [0, 1]
